File: Semester4/Advanced Programming/ThriftProj/JavaThrift/Lab2_Hw/Networking/src/main/java/objProtocol/ComplexRequest.java
package objProtocol;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ComplexRequest implements IRequest, Serializable {

    private List<Object> data;
    private RequestType requestType;

    public ComplexRequest(RequestType requestType){
        this.data = null;
        this.requestType = requestType;
    }

    public ComplexRequest(RequestType requestType, Object[] data){
        this.data = new ArrayList<Object>(Arrays.asList(data));
        this.requestType = requestType;
    }

    @Override
    public String toString() {
        return "ComplexRequest{" +
                "data=" + data +
                ", requestType=" + requestType +
                '}';
    }

    @Override
    public Object getData() {
        return this.data;
    }

    @Override
    public RequestType getRequestType() {
        return requestType;
    }
}

File: Semester4/Advanced Programming/ThriftProj/C#Thrift/Lab2_Festival/Common/protocol/ComplexRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Common.protocol
{
    [Serializable]
    public class ComplexRequest : IRequest
    {
        private List<object> data;
        private RequestType requestType;

        public ComplexRequest(RequestType requestType, params object[] data)
        {
            this.data = new List<object>(data);
            this.requestType = requestType;
        }

        public ComplexRequest(RequestType requestType)
        {
            this.data = null;
            this.requestType = requestType;
        }

        public object GetData()
        {
            return this.data;
        }

        public RequestType GetRequestType()
        {
            return this.requestType;
        }

        public override string ToString()
        {
            if (data == null)
                return "ComplexRequest: {"
                    + " RequestType: " + requestType.ToString() + "}";
            return "ComplexRequest: {"
                + " RequestType: " + requestType.ToString()
                + " Data: " + data.ToString() + "}";
        }
    }
}
